code/modules/mob/living/simple_animal/hostile/headcrab.dm

```diff
@@ -196,3 +203,223 @@
+		if(MODE_REST)		// idle
+			gonarch_rest()
+		if(MODE_RETURNING)		// returning home
+			gonarch_return()
+		if(MODE_NEWHOME)		// Seeking new home
+			gonarch_newhome()
+		if(MODE_RAMPAGE)		// Cannot find nest, frustration at max - rampage
+			gonarch_rampage()
+		if(MODE_DEFENDING) //If it goes down here, there is no target anymore. And it should return.
+			mode_switch(MODE_RETURNING, list('sound/creatures/gonarch_return.ogg'))
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/mode_switch(mode_to_switch, sound_to_play)
+	if(mode_to_switch)
+		mode = mode_to_switch
+		if(sound_to_play)
+			var/sound_picked = pick(sound_to_play)
+			playsound(get_turf(src), sound_picked, 200, 1, 15, null, null, null, FALSE, TRUE)
+	else
+		stack_trace("[src] tried to switch to a NULL mode.")
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_rest()
+	if(get_area(src) == home_nest_area) // If I am home & Everything is calm
+		if(aggro_vision_range != gonarch_aggro_vision_range || vision_range != gonarch_standard_vision_range || speed != gonarch_standard_speed)
+			aggro_vision_range = gonarch_aggro_vision_range
+			vision_range = gonarch_standard_vision_range
+			speed = gonarch_standard_speed
+		if(homesick)
+			homesick = 0
+			step_towards(src, home_nest_turf, speed)
+		make_headcrabs() //Its its own proc, so they can be called by admins.
+	else
+		if(homesick >= 5)
+			mode_switch(MODE_RETURNING, list('sound/creatures/gonarch_return.ogg'))
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_return()
+	speed = -1
+	var/turf/olddist = get_dist(src, home_nest_turf)
+	step_towards(src, home_nest_turf, speed)
+	target = home_nest_turf
+	if((get_dist(src, target)) >= (olddist))
+		frustration++
+		if(frustration > frustration_limit)
+			DestroyPathToTarget()
+			homesick++
+	else
+		frustration = 0
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_newhome()
+	var/list/turfs = new/list()
+	for(var/turf/T in range(src, gonarch_nestfind_range))
+		if(T in range(src, 4))
+			continue
+		if(istype(T, /turf/space))
+			continue
+		if(T.density)
+			continue
+		var/lightingcount = T.get_lumcount(0.5) * 10
+		if(lightingcount > 1 || gonarch_finding_home_attempts >= gonarch_finding_home_limit_attempts)
+			continue
+		turfs += T
+	if(!turfs.len)
+		gonarch_nestfind_range += 5
+		gonarch_finding_home_attempts++
+		return
+	home_nest_turf = pick(turfs)
+	home_nest_area = get_area(home_nest_turf)
+	mode_switch(MODE_RETURNING, list('sound/creatures/gonarch_return.ogg'))
+
+/mob/living/simple_animal/hostile/headcrab/gonarch/proc/gonarch_rampage()
+	//We want to lose all targets, and get the hell out of here. Vision/care is restored when back in base.
+	aggro_vision_range = 0
+	vision_range = 0
+	LoseTarget()
+	homesick = 0
+	for(var/mob/living/carbon/C in hearers(gonarch_screech_range, src))
+		to_chat(C, "<span class='warning'><font size='3'><b>You hear a ear piercing shriek and your senses dull!</font></b></span>")
```

Comment:
Suggestion if it isn't the case already: earmuffs providing protection against this?

Reply:
Kyet was against it.